src/zjs_callbacks.c

```diff
@@ -149,6 +151,22 @@ static zjs_callback_id new_id(void)
     return id;
 }
 
+typedef struct deferred_work {
+    zjs_deferred_work callback;
+    u32_t length;  // length of user data
+    char data[0];  // user data
+} deferred_work_t;
+
+static void deferred_work_callback(void *handle, const void *args) {
+    const deferred_work_t *deferred = (const deferred_work_t *)args;
+
+    if (deferred->callback) {
```

Comment:
Not really; the callback is only used by `zjs_defer_work`, and that always allocates at least the four bytes for the user callback function pointer. Someone could call it wrong later, but that's their bug and problem... and they'll find it when the run on the likes of K64F. :) In this constrained space, it doesn't make sense to have unnecessary NULL checks, though extra checking may be considered good practice in desktop/server programming. IMHO.